src/ConversingUsers.js

```diff
@@ -1,6 +1,36 @@
-const users = Object.freeze({
-  0: "You",
-  1: "Alexa"
+const { Map } = require("immutable");
```

Comment:
This file is confusing. There is lot of redundancy amongst the concepts. For example, you have defined an enum of userIds but you never used the enum to create your users. Why not have something like this, which tightly couples userIds with users?

```js
const UsersEnum = Object.freeze({
  YOU: 0,
  ALEXA: 1,
  MEDIATOR: 2,
  properties: {
    0: {name: "You"},
    1: {name: "Alexa"},
    2: {name: "Mediator"}
  }
});

module.exports = {
  users: UsersEnum
};

// Testing module
it("test users enum", () => {
  console.log("user Id: " + users.YOU); // user Id: 0
  console.log("user Name: " + users.properties[users.YOU].name); // user Name: You
});
```

Reply:
`users.properties[users.YOU].name` doesn't look very intuitive either. I agree that the three level indirection is also not intuitive and the User class is probably an overkill given the use case at hand. 

I removed the User class altogether, coupled the userIds and users by using the former in the latter and cleaned up the tests. 

I also made the Ids more readable because there is not reason for them to be numbers. Since we are strictly controlling the 'users' that exist in the system, having verbose Ids shouldn't hurt and might even help in debugging later on.

Reply:
Okay